boa/src/builtins/string/mod.rs

```diff
@@ -1161,6 +1162,30 @@ impl String {
         RegExp::match_all(&re, this.to_string(ctx)?.to_string())
     }
 
+    /// `String.prototype.search( regexp )`
+    ///
+    /// The `search()` method returns the index of the first match between the [`regular expression`][regex] and the given string, or `-1` if no match was found.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.search
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search
+    /// [regex]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+    pub(crate) fn search(this: &Value, args: &[Value], ctx: &mut Context) -> Result<Value> {
+        let re = match args.get(0) {
+            Some(arg) if !arg.is_null() && !arg.is_undefined() => RegExp::constructor(
```

Comment:
Can use just 1 function to check for both cases of `null` and `undefined`:
https://github.com/boa-dev/boa/blob/2cb24427f1b5e608b453514af413004d3292c2f8/boa/src/value/mod.rs#L311-L313